src/components/Form/index.tsx

```diff
@@ -0,0 +1,86 @@
+import RjsForm from '@rjsf/mui';
+import ajvValidator from '@rjsf/validator-ajv8';
+import { FormProps as RjsFormProps } from '@rjsf/core';
+import { useCallback } from 'react';
+import { Box, Button, ButtonProps } from '@mui/material';
+import { PasswordWidget } from './PasswordWidget';
+import { TextWidget } from './TextWidget';
+import { TextareaWidget } from './TextareaWidget';
+import { SelectWidget } from './SelectWidget';
+import { DateTimeWidget } from './DateTimeWidget';
+import { UpDownWidget } from './UpDownWidget';
+
+const internalWidgets: {
+	[k: string]: any;
+} = {
+	PasswordWidget,
+	EmailWidget: TextWidget,
+	TextWidget,
+	TextareaWidget,
+	SelectWidget,
+	UpDownWidget,
+	DateTimeWidget,
+};
+
+export interface MuiFormProps extends Omit<RjsFormProps, 'validator'> {
+	/** If true, do not display the form submit button */
+	hideSubmitButton?: boolean;
+	/** Properties that are passed to the submit button, these are the same props used for the [`Button`](#button) component */
+	submitButtonProps?: ButtonProps;
+	/** If passed, it will show a secondary button, these are the same props used for the [`Button`](#button) component */
+	secondaryButtonProps?: ButtonProps;
+
+	validator?: RjsFormProps['validator'];
+}
+
+export const MuiForm: React.FC<React.PropsWithChildren<MuiFormProps>> = ({
+	hideSubmitButton,
+	submitButtonProps,
+	secondaryButtonProps,
+	validator = ajvValidator,
+	widgets,
+	fields,
+	children,
+	...otherProps
+}) => {
+	const handleKeyUpDown = useCallback(
+		(e: React.KeyboardEvent<HTMLDivElement>) => {
+			if (e.key === 'Enter' && !hideSubmitButton) {
+				e.stopPropagation();
+			}
+		},
+		[hideSubmitButton],
+	);
+	return (
+		<Box onKeyDown={handleKeyUpDown} onKeyUp={handleKeyUpDown}>
+			<RjsForm
+				validator={validator}
+				showErrorList={false}
+				onChange={(data) => {
+					otherProps.onChange?.(data);
+				}}
+				widgets={{ ...internalWidgets, ...(widgets || {}) }}
+				{...otherProps}
+			>
+				{hideSubmitButton && <span />}
+
+				{!hideSubmitButton && (
+					<Button
+						{...submitButtonProps}
+						color="customBlue"
+						variant="contained"
+						type="submit"
+					/>
+				)}
+				{!!secondaryButtonProps && (
+					<Button
+						{...secondaryButtonProps}
+						variant="outlined"
+						color="customBlue"
+					/>
+				)}
```

Comment:
What's the case where we would have this?
I see that we use submitButtonProps and submitButtonText, but the only use I see of this is on the Profile setup page. I don't believe a form will generally have a second button (maybe there's some standard UX rules/requirements I'm missing here cc @drskullster, the only exception I can think of is a multi-page form where the secondary button might be to go back a page). Arguably the Profile setup page can become a `DialogWithCloseButton` or feature some other way to skip (i.e. just change the Submit button text to `Skip` if the inputs are unchanged maybe? 🤷 ) and we can avoid this extremely rare extra prop

Reply:
>  I don't believe a form will generally have a second button (maybe there's some standard UX rules/requirements I'm missing here cc @drskullster, the only exception I can think of is a multi-page form where the secondary button might be to go back a page). 

Another example I can imagine is a `Reset to default` button, which would indeed be a valid usecase. But in terms of how we already use this prop, I'm not sure a `Skip` button is standard UX for a Form. Could be wrong though, I can see it go either way now tbh but I want us to think about it before just sticking with what we had last time

Reply:
@myarmolinsky I was keeping as a first step the logic we have in rendition to not make a complete change in a single PR. Ideally we should have a buttons.map here, not be opinionated. A form might have a clear button, a validate button, a submit button...  I'm trying to not focus on our single use case.